c/cert/src/rules/CON34-C/AppropriateThreadObjectStorageDurations.ql

```diff
@@ -0,0 +1,46 @@
+/**
+ * @id c/cert/appropriate-thread-object-storage-durations
+ * @name CON34-C: Declare objects shared between threads with appropriate storage durations
+ * @description Accessing thread-local variables with automatic storage durations can lead to
+ *              unpredictable program behavior.
+ * @kind problem
+ * @precision high
+ * @problem.severity error
+ * @tags external/cert/id/con34-c
+ *       correctness
+ *       concurrency
+ *       external/cert/obligation/rule
+ */
+
+import cpp
+import codingstandards.c.cert
+import codingstandards.cpp.Concurrency
+import semmle.code.cpp.dataflow.TaintTracking
+import semmle.code.cpp.dataflow.DataFlow
+
+class MallocFunctionCall extends FunctionCall {
+  MallocFunctionCall() { getTarget().getName() = "malloc" }
+}
+
+from C11ThreadCreateCall tcc, StackVariable sv, Expr arg, Expr acc
```

Comment:
silly question, is `StackVariable` really the type we want to use here? the rule description says: "Accessing the automatic or **thread-local variables**" and the implementation of `StackVariable` says: `not this.isThreadLocal()`

Reply:
@knewbury01 -- I could be mistaken, but specifically I am trying to identify unsafe variables:

```
not this.isStatic() and
    not this.isThreadLocal()
```

I think is exactly what we want here, since static thread local variables ARE permitted.

Reply:
hm I think I was thinking there was a way that this would miss just thread local variables, that are not static, but I cannot come up with a test case so it seems fine